src/lib/components/Modal/index.tsx

```diff
@@ -74,18 +78,15 @@ const ModalComponent: FC<ModalProps> = ({
         <ModalContext.Provider value={{ popup, onClose }}>
           <div
             aria-hidden={!show}
-            className={classNames(
-              'fixed top-0 right-0 left-0 z-50 h-modal overflow-y-auto overflow-x-hidden md:inset-0 md:h-full',
-              placementClasses[placement],
-              {
-                'flex bg-gray-900 bg-opacity-50 dark:bg-opacity-80': show,
-                hidden: !show,
-              },
-            )}
+            className={classNames(theme.base, placementClasses[placement], {
+              [theme.visible]: show,
```

Comment:
IMO this makes more sense as theme.show.on/off so you dont have two states visible and hidden which are named differently from the Modal API (show). Nitpick

Reply:
Works for me.